plutus-core/cost-model/create-cost-model/CreateBuiltinCostModel.hs

```diff
@@ -726,3 +731,21 @@ mkNilPairData cpuModelR = do
   let memModel = ModelOneArgumentConstantCost 32
   pure $ CostingFun cpuModel memModel
 -- () -> [] :: [(Data,Data)]
+
+---------------- Misc constructors ----------------
+
+showInteger :: MonadR m => (SomeSEXP (Region m)) -> m (CostingFun ModelOneArgument)
+showInteger cpuModelR = do
+  cpuModel <- ModelOneArgumentLinearCost <$> readModelLinearInX cpuModelR
+  let memModel =
+        -- A 64-bit Integer has up to 21 characters when shown (-(2^64-1))
+        ModelOneArgumentLinearCost $ ModelLinearSize 0 21
+  pure $ CostingFun cpuModel memModel
+
+showByteString :: MonadR m => (SomeSEXP (Region m)) -> m (CostingFun ModelOneArgument)
+showByteString cpuModelR = do
+  cpuModel <- ModelOneArgumentLinearCost <$> readModelLinearInX cpuModelR
+  let memModel =
+        -- A ByteString of length 8 has 10 characters when shown (one per element, plus quotes)
+        ModelOneArgumentLinearCost $ ModelLinearSize 0 10
```

Comment:
Maybe `ModelLinearSize 2 8`, since the two quotes have a constant cost of 2 units?

Reply:
> -- A ByteString of length 8 has 10 characters when shown (one per element, plus quotes)
Does it though?   
T.pack $ show (B.pack [0..32])
"\"\\NUL\\SOH\\STX\\ETX\\EOT\\ENQ\\ACK\\a\\b\\t\\n\\v\\f\\r\\SO\\SI\\DLE\\DC1\\DC2\\DC3\\DC4\\NAK\\SYN\\ETB\\CAN\\EM\\SUB\\ESC\\FS\\GS\\RS\\US \""

```
Does that matter though?  I'm actually not sure: all those extra backlashes are confusing me and I'm not even sure how much room a `Char` takes up .  Also, the memory usage is in 8-byte words (although I really wish it was bytes) so maybe that's a safe overestimate.

Reply:
You are right regarding 64 characters to display per 8 bytes. I didn't know the output is measured by 8-bytes words. If so, then the slope should be 8.